File: opencv/Task1_basic_operates/core/phone.py
import cv2

# 读取图片
img = cv2.imread('all.jpg')

# 假设手机部分在图片的左上角
x, y, w, h = 187, 286, 295, 113
phone_img = img[y:y+h, x:x+w]

# 保存截取的部分
cv2.imwrite('phone.jpg', phone_img)

File: EmbemdedLinux/Task4_OpenCV_Cpp/Code/phone.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

int main() {
    cv::Mat img = cv::imread("/opencv.cpp/all.jpg");
    if (img.empty()) {
        std::cout << "Unable to read picture" << std::endl;
        return -1;
    }

    int x1 = 311, y1 = 92, x2 = 478, y2 = 440;
    cv::Mat phonePart = img(cv::Rect(x1, y1, x2 - x1, y2 - y1));

    if (!cv::imwrite("phone.jpg", phonePart)) {
        std::cout << "Unable to save picture" << std::endl;
        return -1;
    }

    std::cout << "Image saved successfully" << std::endl;
    return 0;
}